packages/design-picker/src/components/generated-design-picker.tsx

```diff
@@ -115,14 +118,39 @@ const GeneratedDesignPicker: React.FC< GeneratedDesignPickerProps > = ( {
 	verticalId,
 	locale,
 	heading,
+	footer,
+	isStickyFooter,
 	onPreview,
 	onViewMore,
 } ) => {
 	const { __ } = useI18n();
+	const headingRef = useRef< HTMLDivElement >( null );
+	const observerRef = useRef< IntersectionObserver >();
+	const [ isHeadingInViewport, setIsHeadingInViewport ] = useState( true );
+
+	useEffect( () => {
+		if ( ! headingRef.current || observerRef.current || ! isStickyFooter ) {
+			return;
+		}
+
+		const handler = ( entries: IntersectionObserverEntry[] ) => {
+			if ( ! entries.length ) {
+				return;
+			}
+
+			const [ entry ] = entries;
+			setIsHeadingInViewport( entry.isIntersecting );
+		};
+
+		observerRef.current = new IntersectionObserver( handler, { threshold: 1 } );
```

Comment:
I think we have to specify the `rootMargin` here.

```
rootMargin: "48px 0"
```

Reply:
From [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API):
> The default is 0 (meaning as soon as even one pixel is visible, the callback will be run). A value of 1.0 means that the threshold isn't considered passed until every pixel is visible.

I changed the threshold to the default value of 0. Technically, this might cause 12px of no-CTA button due to it being vertically center-aligned, but I think that's acceptable 😅

Reply:
I still prefer to show the sticky footer only when the “Continue” button at least is invisible if possible 🥲

Reply:
In that case, I've change to observe the button rather than the heading.